custom/icds_reports/reports/disha.py

```diff
@@ -81,32 +81,32 @@ def _write_data_in_chunks(self, temp_path):
         #   }
         columns = self._get_columns()
         indicators = self._get_rows()
-        metadata_line = '{{'\
+        metadata_line = u'{{'\
             '"month":"{month}", '\
             '"state_name": "{state_name}", '\
             '"column_names": {columns}, '\
             '"rows": ['.format(
                 month=self.month,
                 state_name=self.state_name,
-                columns=json.dumps(columns, ensure_ascii=False))
-        with open(temp_path, 'w', encoding='utf-8') as f:
-            f.write(metadata_line)
-            written_count = 0
-            num_batches = 10
-            for count, (_, end, total, chunk) in enumerate(batch_qs(indicators, num_batches=num_batches)):
-                chunk_string = json.dumps(list(chunk), ensure_ascii=False)
-                # chunk is list of lists, so skip enclosing brackets
-                f.write(chunk_string[1:-1])
-                written_count += len(chunk)
-                if written_count != total:
-                    f.write(",")
-                logger.info("Processed {count}/{total} batches".format(count=count, total=num_batches))
-            f.write("]}")
+                columns=json.dumps(columns, ensure_ascii=False)).encode('utf8')
+        file_obj.write(metadata_line)
+        written_count = 0
+        num_batches = 10
+        for count, (_, end, total, chunk) in enumerate(batch_qs(indicators, num_batches=num_batches)):
+            chunk_string = json.dumps(list(chunk), ensure_ascii=False).encode('utf8')
```

Comment:
The return type of `json.dumps(...)` is different on Python 2 and 3 (due to a bug in Python 2). This code will work on both Python 2 and 3, but it's kinda sneaky to do an unnecessary encode on Python 2 like that since the result is already bytes (see also [this comment](https://github.com/dimagi/langcodes/pull/19#discussion_r189239788)).

My recommendation here would be to revert the changes in this file and make a simpler change here:
```py
chunk_string = json.dumps(list(chunk), ensure_ascii=False)
if six.PY2:
    chunk_string = chunk_string.decode('utf-8')
```
With that change all the code should just work as it was written before, using `open(temp_path, 'w', encoding='utf-8')`. The `six.PY2` check + decode could eventually be removed when we switch to Python 3 (although it will not hurt to let it there).

Alternately, if you want to keep the code as you have it (with some fixes I'm suggesting in other comments), you could do this instead:
```py
chunk_string = json.dumps(list(chunk), ensure_ascii=False)
if six.PY3:
    chunk_string = chunk_string.encode('utf-8')
```